### PR TITLE
Set up `pre-commit`, re-run linters

### PR DESCRIPTION
**Pull request by mattwthompson**

### CODE REVIEW

**Comment by jthorton** [comment]
@mattwthompson thanks for starting this it would be great to get these fixes in! is this ready to go?

File: setup.cfg [hunk 1]
```diff
@@ -20,6 +20,10 @@ exclude_lines =
 max-line-length = 88
 ignore = E203, E266, E501, W503
 select = B,C,E,F,W,T4,B9
+per-file-ignores =
+    openff/interchange/_tests/unit_tests/test_types.py:F821
```
[the remark this answers is not in the document]
Comment (mattwthompson) [reply setup.cfg:24]: Whoops

**Comment by mattwthompson** [comment]
Should be good, yeah - the linters appear to be passing now, even though I'm not sure CI will

**Comment by mattwthompson** [comment]
This, #212, and #220 are in some sort of holding pattern for getting tests passing based off of the `main` branch. I'm not sure the best way to proceed; this and #212 are probably standalone automation improvements that could be passed without tests passing, but I'm less than thrilled to see the tests time out or fail so often

**Review by jthorton — APPROVED** [review]

**Pull request merged**